plugin/trino-pulsar/src/main/java/io/trino/plugin/pulsar/PulsarConnector.java

```diff
@@ -0,0 +1,112 @@
+ * under the License.
+ */
+package io.trino.plugin.pulsar;
+
+import io.airlift.bootstrap.LifeCycleManager;
+import io.airlift.log.Logger;
+import io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata;
+import io.trino.spi.connector.Connector;
+import io.trino.spi.connector.ConnectorMetadata;
+import io.trino.spi.connector.ConnectorRecordSetProvider;
+import io.trino.spi.connector.ConnectorSplitManager;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.transaction.IsolationLevel;
+
+import javax.inject.Inject;
+
+import static io.trino.spi.transaction.IsolationLevel.READ_COMMITTED;
+import static io.trino.spi.transaction.IsolationLevel.checkConnectorSupports;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This file contains implementation of the connector to the Trino.
+ */
+public class PulsarConnector
+        implements Connector
+{
+    private static final Logger log = Logger.get(PulsarConnector.class);
+
+    private final LifeCycleManager lifeCycleManager;
+    private final PulsarMetadata metadata;
+    private final PulsarSplitManager splitManager;
+    private final PulsarRecordSetProvider recordSetProvider;
+    private final PulsarConnectorConfig pulsarConnectorConfig;
+
+    @Inject
+    public PulsarConnector(
+            LifeCycleManager lifeCycleManager,
+            PulsarMetadata metadata,
+            PulsarSplitManager splitManager,
+            PulsarRecordSetProvider recordSetProvider,
+            PulsarConnectorConfig pulsarConnectorConfig)
+    {
+        this.lifeCycleManager = requireNonNull(lifeCycleManager, "lifeCycleManager is null");
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.splitManager = requireNonNull(splitManager, "splitManager is null");
+        this.recordSetProvider = requireNonNull(recordSetProvider, "recordSetProvider is null");
+        this.pulsarConnectorConfig = requireNonNull(pulsarConnectorConfig, "pulsarConnectorConfig is null");
+    }
+
+    @Override
+    public ConnectorTransactionHandle beginTransaction(IsolationLevel isolationLevel, boolean readOnly)
+    {
+        checkConnectorSupports(READ_COMMITTED, isolationLevel);
+        return PulsarTransactionHandle.INSTANCE;
+    }
+
+    @Override
+    public ConnectorMetadata getMetadata(ConnectorTransactionHandle transactionHandle)
+    {
+        return new ClassLoaderSafeConnectorMetadata(metadata, getClass().getClassLoader());
+    }
+
+    @Override
+    public ConnectorSplitManager getSplitManager()
+    {
+        return splitManager;
+    }
+
+    @Override
+    public ConnectorRecordSetProvider getRecordSetProvider()
+    {
+        return recordSetProvider;
+    }
+
+    public void initConnectorCache() throws Exception
+    {
+        PulsarConnectorCache.getConnectorCache(pulsarConnectorConfig);
+    }
+
+    @Override
+    public final void shutdown()
+    {
+        try {
+            this.pulsarConnectorConfig.close();
```

Comment:
not needed `this.` prefix (please remove it from other places too).

Reply:
updated.